src/NetscriptFunctions/Corporation.ts

```diff
@@ -803,6 +786,20 @@ export function NetscriptCorporation(): InternalAPI<NSCorporation> {
         CorporationPromise.promise = new Promise<CorpStateName>((res) => (CorporationPromise.resolve = res));
       return CorporationPromise.promise;
     },
+    restartCorporation:
+      (ctx) =>
+      (_corporationName, _selfFund): boolean => {
+        checkAccess(ctx);
+        const corporationName = helpers.string(ctx, "corporationName", _corporationName);
+        const selfFund = !!_selfFund;
+        return createCorporation(corporationName, selfFund, true);
```

Comment:
I think it's a bit of an API smell to both return true/false *and* throw exceptions for similar types of issues. (For instance, throw if the BN doesn't support corps, but return false if you didn't provide a name: the logical distinction there is arbitrary.)

I know some people don't like dealing with exceptions, but IMO when you call this method you *always* expect it to work, and there's enough different ways for it to fail that the descriptive message of an exception is helpful.